### ISSUE TITLE
Did not work on my P65xRP6

### ISSUE THREAD

**Issue by alwzying** [datasone/ClevoControl#5]
Did not work on my laptop. I compiled the kext with latest source code and I checked that the kext had been loaded.

**Comment by datasone** [comment]
Did you add corresponding codes into DSDT as [README](https://github.com/datasone/ClevoControl/blob/master/README.md) said?
It's a workaround to cannot directly attaching WMI devices.(This kext just call corresponding functions in WMI device)

**Comment by alwzying** [comment]
I have, maybe different model needs different code?

**Comment by datasone** [comment]
Well, as I have known all CLEVO full color keyboard should share the same code.
I recommend you to check your DSDT to see if it has the OEM WMI device because many hackintosh guides suggest patching DSDT to remove OEM WMI device.

**Comment by alwzying** [comment]
<img width="743" alt="2018-07-30 2 07 29" src="https://user-images.githubusercontent.com/26023755/43380038-f5c9ceb0-9401-11e8-91e3-31a9d68d6e18.png">
Is this right?

**Comment by alwzying** [comment]
[DSDT.dsl.zip](https://github.com/datasone/ClevoControl/files/2239970/DSDT.dsl.zip)
and this is my dsdt

**Comment by datasone** [comment]
It seems there is no problem in DSDT, at least the WMI part...
I've noticed there is no _OSI fake in DSDT, as it's related to EC initialization, you can try it out.(i.e. replace `If (_OSI ("Windows 2015"))` to `If (LOr (_OSI ("Darwin"), _OSI ("Windows 2015")))` )

**Comment by alwzying** [comment]
Still cannot work. And should I replace PNP0C02 with PNP0C14 as the picture shows?

**Comment by datasone** [comment]
No, the `SMCD` device is only a wrapper. And as [here](https://github.com/datasone/ClevoControl/blob/780123fa6efc06961f826a9c7c83b8f73b5c2eaa/ClevoControl/Info.plist#L31) said, the kext attaches on device named "MON0000"(or "MON00000").
And this name is only used when loading the kext, as the kext is loaded, this shouldn't be a problem.

**Comment by datasone** [comment]
To figure out where the problem is, you may need to play with ACPI under Linux or Windows.
As there's no such info in your profile, how can I contact you?

**Comment by alwzying** [comment]
It seems like we are all Chinese...My QQ: 857517669 or WeChat:15301938691
